### DISCUSSION TITLE
What's the difference between skip_dir and sync_list? Why should I use one over the other?

### DISCUSSION THREAD

**Discussion by pepper-lees** [abraunegg/onedrive#2368] [Q&A] [answered]
I see discussion in the issues that when people use sync_list, it causes the sync process to be a lot slower. Can I prevent that from simply skipping the directories I don't want to sync via skip_dir? What's the difference between the two? I think it would be helpful to have the answer to this question either in USAGE.md or the FAQ page. If the syncing process **does** end up faster by using skip_dir instead of sync_list, then I think it would be beneficial to include that information somewhere easy to see, maybe including a tips section explaining what you can do to make sure syncing is fast.

**Comment by abraunegg** [comment, marked as answer]
@JL102 
> I see discussion in the issues that when people use sync_list, it causes the sync process to be a lot slower.

It is only 'slower' in their mind - as - there is technically zero difference in processing between using sync_list or not. The difference is a boolean comparison as to whether include the item or to exclude. Because of the logging output however - this is where the perception comes from that it is slower - however processing wise it is not.

> Can I prevent that from simply skipping the directories I don't want to sync via skip_dir?

Yes you can

> What's the difference between the two?

Technically zero difference, except when using 'sync_list' there is a more robust check being done to ensure that changes (moves|additions|renames) made online to an item that was maybe previously excluded now included or reverse is correctly attributed and handled.

From a processing perspective there is zero difference.
